InventoryInStorePickup/Model/SearchRequest/Area/SearchTerm/PostcodeParser.php

```diff
@@ -24,30 +25,47 @@ class PostcodeParser implements ParserInterface
      */
     private $validator;
     /**
-     * @var DelimiterParser
+     * @var DelimiterConfig
      */
-    private $parser;
+    private $delimiterConfig;
 
     /**
      * @param ValidatorInterface $validator
-     * @param DelimiterParser $parser
+     * @param DelimiterConfig $delimiterConfig
      */
-    public function __construct(ValidatorInterface $validator, DelimiterParser $parser)
+    public function __construct(ValidatorInterface $validator, DelimiterConfig $delimiterConfig)
     {
         $this->validator = $validator;
-        $this->parser = $parser;
+        $this->delimiterConfig = $delimiterConfig;
     }
 
     /**
      * @inheritdoc
      */
     public function execute(string $searchTerm, DataObject $dataObject): void
     {
-        $searchQuery = $this->parser->getSearchQuery($searchTerm);
+        $searchQuery = $this->getSearchQuery($searchTerm);
         if ($this->validator->validate($searchQuery, $dataObject->getData(self::COUNTRY))) {
             $dataObject->setData(self::POSTCODE, $searchQuery);
         } else {
             $dataObject->setData(self::POSTCODE, '');
         }
     }
+
+    /**
+     * Parse and return search query from search term.
+     *
+     * @param string $searchTerm
+     *
+     * @return string
+     */
+    public function getSearchQuery(string $searchTerm): string
```

Comment:
This method is duplicated 3 times. Something went wrong.

Reply:
2 times*
It was agreed to do not use "helper" classes.